warmup.py

```diff
@@ -240,17 +247,42 @@ def accumulate(self, L, points):
         The sample is accumulated into the radial distribution average.
 
         """
-        pass
+        if (self.rmax > (L / 2)):
+            raise ValueError('rmax > L/2')
+        else:
+            dist = np.zeros(sum(range(len(points))))                                                               #array of distance magnitudes between all point combinations
+            for i, point in enumerate(points):
+                neighbor_index = i + 1
+                point_shifted = point - (L * np.round(np.array(point) / L))                                        #wrap origin particle to periodic boundaries
+                while neighbor_index < len(points):
+                    neighbor = points[neighbor_index]                                                              #array of all other particles to compare to origin particle
+                    dist_vector = np.zeros_like(neighbor)                                                          #array of distances in x, y, z directions
+                    for j, (coord_neighbor, coord_point_shifted) in enumerate(zip(neighbor, point_shifted)):
+                        dist_vector[j] = coord_neighbor - coord_point_shifted - (L * np.round(coord_neighbor / L)) #wrap dist_vector to periodic boundaries
+                    dist[i] = np.linalg.norm(dist_vector)
+                    neighbor_index += 1
+            for i, bin_center in enumerate(self.centers):
+                deltaN = 0                                                                                         #particle count within shell
+                for dist_i in dist:
+                    radius_outer = self.edges[i + 1]
+                    radius_inner = self.edges[i]
+                    if radius_inner < dist_i <= (radius_outer + 1e-8):
+                        deltaN += 1
+                if radius_inner != 0:
+                    deltaV = (4 / 3) * np.pi * ((radius_outer ** 3) - (radius_inner ** 3))                         #shell volume
+                    self.rdf_var[i] = deltaN / deltaV
+            rho = len(points) / (L ** 3)                                                                           #overall particle number density
+            self.rdf_var /= rho
 
     def reset(self):
         r"""Reset the accumulator.
 
         The sample accumulator is reset to zeros.
 
         """
-        pass
+        self.rdf_var = np.zeros(len(self.centers))
```

Comment:
In addition to clearing the internal RDF accumulator, you might want to add a counter keeping track of how many RDF "samples" you have added into it. Then, the average RDF is `self._rdf / samples`.